common/djangoapps/philu_commands/management/commands/tests/test_auto_generate_certificates_for_open_courses.py

```diff
@@ -0,0 +1,133 @@
+    """
+    Tests for `auto_generate_certificates_for_open_courses.py` command.
+    """
+
+    @mute_signals(signals.pre_save, signals.post_save)
+    def setUp(self):
+        """
+        This function is responsible for creating courses for every test and mocking the function for tests.
+        """
+        super(TestAutoGenerateCertificateForOpenCourse, self).setUp()
+        self.course_1 = CourseFactory.create(display_name='test course 1', run='Testing_course_1')
+        self.course_1.end = datetime.now(pytz.UTC) - timedelta(hours=2)
+        self.user = UserFactory(username="test", email="test@example.com", password="123")
+
+    @patch(
+        'philu_commands.management.commands.auto_generate_certificates_for_open_courses._is_eligible_for_certificate',
+        return_value=False)
+    @patch('philu_commands.management.commands.auto_generate_certificates_for_open_courses.'
+           'is_course_valid_for_certificate_auto_generation', return_value=True)
+    @patch('philu_commands.management.commands.auto_generate_certificates_for_open_courses.modulestore')
+    @patch(
+        'philu_commands.management.commands.auto_generate_certificates_for_open_courses.CourseEnrollment.objects.filter'
+    )
+    @patch('philu_commands.management.commands.auto_generate_certificates_for_open_courses._get_cert_data')
+    @patch('philu_commands.management.commands.auto_generate_certificates_for_open_courses.generate_user_certificates',
+           return_value="generating")
+    @patch('philu_commands.management.commands.auto_generate_certificates_for_open_courses.log.info')
+    def test_successfully_generate_certificate(
+        self,
+        mock_log_info,
+        mock_generate_user_certificates,
+        mock__get_cert_data,
+        mock_course_enrollments,
+        mock_modulestore,
+        _mock_is_course_valid,
+        _mock__is_eligible_for_certificate
+    ):
+        """
+        Test 'Successfully generate certificate'
+        """
+        mock_response = Mock(name="mock module store", **{"get_courses.return_value": [self.course_1]})
+        mock_modulestore.return_value = mock_response
+        mock_objects = CourseEnrollmentFactory.create(
+            user=self.user, course_id=self.course_1.id, mode='honor')
+        mock_course_enrollments.return_value = Mock(**{"all.return_value": [mock_objects], "get_items": {}})
+        mock__get_cert_data.return_value = Mock(**{"cert_status": "requesting"})
+
+        def assert_certificates_generated(info):
+            """
+            assert certification generation
+            """
+            certificate_success_message = 'Generating certificate for user with ' \
+                                          'username: {} and user_id: {} with ' \
+                                          'generation status: {}'.format(self.user.username, self.user.id,
+                                                                         mock_generate_user_certificates())
```

Comment:
This can be written in a better way. Maybe we can keep the msg string in a variable like "**success_msg**" and then use this variable at this line and "**success_msg.format(self.user.username, self.user.id, mock_generate_user_certificates())**". Line breaks will look a-lot more cleaner this way.